operators/pkg/controller/common/settings/canonical_config.go

```diff
@@ -0,0 +1,290 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package settings
+
+import (
+	"fmt"
+	"reflect"
+	"sort"
+	"strconv"
+	"strings"
+
+	ucfg "github.com/elastic/go-ucfg"
+	udiff "github.com/elastic/go-ucfg/diff"
+	uyaml "github.com/elastic/go-ucfg/yaml"
+	"github.com/pkg/errors"
+	yaml "gopkg.in/yaml.v2"
+)
+
+// CanonicalConfig contains configuration for an Elastic resource ("elasticsearch.yml" or "kibana.yml"),
+// as a hierarchical key-value configuration.
+type CanonicalConfig ucfg.Config
+
+// Options are config options for the YAML file. Currently contains only support for dotted keys.
+var Options = []ucfg.Option{ucfg.PathSep(".")}
+
+// NewCanonicalConfig creates a new empty config.
+func NewCanonicalConfig() *CanonicalConfig {
+	return fromConfig(ucfg.New())
+}
+
+// NewCanonicalConfigFrom creates a new config from the API type.
+func NewCanonicalConfigFrom(data untypedDict) (*CanonicalConfig, error) {
+	config, err := ucfg.NewFrom(data, Options...)
+	if err != nil {
+		return nil, err
+	}
+	return fromConfig(config), nil
+}
```

Comment:
I updated the argument of `NewCanonicalConfigFrom` to remove the dependency to `estype.Config`.
https://github.com/elastic/cloud-on-k8s/blob/f00602603757aed48c15da96c557b40edb9a4dbe/operators/pkg/controller/elasticsearch/settings/canonical_config.go#L33-L39